src/sync/mod.rs

```diff
@@ -1 +1,4 @@
+//! Synchronization primitives.
+
 pub mod list;
```

Comment:
I agree with @Vtec234 here - let's keep it simple and focus on EBR at this point.

`MsQueue`, `TreiberStack`, linked lists, etc. are pretty easy data structures to (re)implement, so I don't see much value in abstracting them away. There will be a time for that later on. :)